x-pack/plugins/observability/public/components/co_pilot_prompt/co_pilot_prompt_feedback.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiText,
+  useEuiTheme,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import type { ChatCompletionRequestMessage } from 'openai';
+import React, { useCallback, useEffect, useState } from 'react';
+import { CoPilotPromptId } from '../../../common';
+import type { CoPilotService } from '../../typings/co_pilot';
+
+interface Props {
+  coPilot: CoPilotService;
+  promptId: CoPilotPromptId;
+  messages?: ChatCompletionRequestMessage[];
+  response: string;
+  responseTime: number;
+}
+
+export function CoPilotPromptFeedback({
+  coPilot,
+  promptId,
+  messages,
+  response,
+  responseTime,
+}: Props) {
+  const theme = useEuiTheme();
+
+  const [hasSubmittedFeedback, setHasSubmittedFeedback] = useState(false);
+
+  const submitFeedback = useCallback(
+    (positive: boolean) => {
+      setHasSubmittedFeedback(true);
+      if (messages) {
+        coPilot
+          .track({
+            messages,
+            response,
+            responseTime,
+            promptId,
+            feedbackAction: positive ? 'thumbsup' : 'thumbsdown',
+          })
+          .catch((err) => {});
+      }
+    },
+    [coPilot, promptId, messages, response, responseTime]
+  );
+
+  const [hasSubmittedTelemetry, setHasSubmittedTelemetry] = useState(false);
+
+  useEffect(() => {
+    if (!hasSubmittedTelemetry && messages) {
+      setHasSubmittedTelemetry(true);
+      coPilot
+        .track({
+          messages,
+          response,
+          responseTime,
+          promptId,
+        })
+        .catch((err) => {});
+    }
+  }, [coPilot, promptId, messages, response, responseTime, hasSubmittedTelemetry]);
+
+  useEffect(() => {
+    setHasSubmittedFeedback(false);
+  }, []);
```

Comment:
I wonder why you need this `useEffect` as you initialized the hasSubmittedFeedback with `false` already.